### DISCUSSION TITLE
Attempting to import from Heroku Postgres v14 to Supabase

### DISCUSSION THREAD

**Discussion by mikecann** [supabase/supabase#8848] [Questions] [answered]
Hi,

We are trying to migrate from Heroku Postgres to Supabase but are suffering some issues that I suspect are related to the fact that our Heroku Postgres server is running v14.4 of postgres and I believe Supabase is running v12 (is that right)?

Is there anything we can do here?

Cheers,

**Comment by GaryAustin1** [comment, marked as answer]
Supabase is Postgres 14 for awhile now.  
Current instance shows 14.1 for `select version()`

**Reply by mikecann** [reply]
ah okay thanks, must be something else going wrong there then. ill investigate

**Comment by Rubenburdin** [comment]
Jumping in on this in 2026, Gary was right that Supabase has been on Postgres 14 for a while (they're actually on Postgres 15 now for new projects). So the version mismatch wasnt your issue. For anyone finding this thread with similar import problems, the most common causes when doing pg_dump from Heroku and pg_restore into Supabase are extension conflicts and ownership issues. Heroku enables certain extensions by default that Supabase might not have enabled yet, and the dump includes role references that dont exist in Supabase. Running the restore with --no-owner --no-acl flags usually clears up most of those errors (https://supabase.com/docs/guides/database/connecting-to-postgres).

Also worth checking if your Heroku dump includes any Heroku specific schemas or extensions like heroku_ext that wont exist on Supabase. You can filter those out during the dump with --exclude-schema=heroku_ext or just ignore the errors on restore since they're usually non critical.

For teams going through this migration who were also using Heroku Connect for Salesforce to Postgres sync, thats the part that breaks completely when you leave Heroku since it only works within their ecosystem. We've helped a lot of teams replace Heroku Connect through [Stacksync](https://www.stacksync.com/heroku-connect-alternative) (https://supabase.com/partners/integrations/stacksync) with real time two way Salesforce to Postgres sync that works with Supabase or any Postgres host, without the locked in infrastructure or the aggressive pricing that scales up as your data grows.

Hope this helps anyone going through the same migration!
